### PR TITLE
Ed1d1a8d patch endDate

### PR DESCRIPTION
**Pull request by ed1d1a8d**
Fix endDate logic so that returned data is in range [startDate, endDate] (inclusive for both days).
Resolves issue https://github.com/sipb/sipb-door/issues/5

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -83,6 +83,7 @@ function visualControl($scope, $filter, $http) {
       r = {};
       r.startDate = Date.parse($filter('date')(d.startDate, 'dd/MMM/yyyy HH:mm:ss')) / 1000;
       r.endDate = Date.parse($filter('date')(d.endDate, 'dd/MMM/yyyy HH:mm:ss')) / 1000;
+      r.endDate += daySeconds; // Capture entire day
       if (r.startDate >= r.endDate) {
```
Comment (ongmingyang) [inline script.js]: If we want the dates to be inclusive of the end interval, could you change this line also to 

```
if (r.startDate > r.endDate) {
```

Thanks so much

File: data.cgi [hunk 2]
```diff
@@ -26,6 +26,8 @@ for line in f:
   if '#' in line:
     continue
   nline=line.split(',')
+  if int(nline[1]) > int(endDate):
+    break
```
Comment (ongmingyang) [inline data.cgi]: good catch, thanks!

**Comment by ongmingyang** [comment]
Hey, thanks for finding and fixing bugs in my very old and cringeworthy code. Could you squash your commits so that I can merge them?

**Comment by ed1d1a8d** [comment]
Squashed.

**Comment by ongmingyang** [comment]
Could you change line 87 in `script.js` to 

```
if (r.startDate > r.endDate) {
```

also?

**Comment by ed1d1a8d** [comment]
I don't think this produces the intended result.
Note that we do `r.endDate += daySeconds` first.
If our comparison was `r.startDate > r.endDate`,
then the program would allow the user to input a
start date that happens one day after the end date.
But this should be invalid.

**Comment by ongmingyang** [comment]
You are right. `daySeconds` is added first. Thanks for the clarification!

**Pull request merged**